test/distributed/test_device_mesh.py

```diff
@@ -454,6 +454,46 @@ def test_all_gather_uneven(self):
             self.assertEqual(all_gathered_tensor.size(), tensor_to_split.size())
             self.assertEqual(all_gathered_tensor, tensor_to_split)
 
+    @with_comms
+    def test_reduce_scatter_contiguous(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        my_rank = device_mesh.get_rank()
+
+        # Init the tensor
+        step = self.world_size * 2
+        total_elem = step**2
+        tensor = torch.arange(0, total_elem).view(step, -1).to(device=self.device_type)
+        tensor = tensor * (my_rank + 1)
+
+        # Get non-contiguous tensor by slicing
+        tensor_to_reduce = tensor[::2, :2]
+        tensor_contiguous = tensor_to_reduce.clone().contiguous()
+
+        # Partial to Shard to trigger reduce_scatter
+        partial_placement = _Partial()
+        shard_spec = Shard(0)
+        new_tensor = partial_placement._to_shard(
```

Comment:
I think we should test the public API `from_local -> redistribute` instead, directly testing private APIs is fragile as it's not testing the user facing API and we might change private APIs frequently and introduces refactor burden